### PR TITLE
add named_parameters that yield name and value of parameters

### PR DESCRIPTION
**Pull request by soumith**
This is useful when having to filter out 'bias' for weight decay (for example)

### CODE REVIEW

**Comment by soumith** [comment]
I wonder if giving full names like we return in `state_dict` is useful. For example filtering all biases of `BatchNorm` is a genuine use-case and isn't covered by this PR.

Should I make this return full names? or guard full names behind a kwarg?

**Comment by apaszke** [comment]
Yeah I think full names would be better, because that's the default for `named_modules`

**Review by apaszke — APPROVED** [review]
